### PR TITLE
feat: implement getWorkflowResultsByInstanceId (V2)

### PR DESCRIPTION
**Pull request by rhkp**
Hi Gloria,
Please find the implementation of getWorkflowResultsByInstanceId (V2). Let me know what do you think?

Thanks,
Kamlesh.

**curl**
curl --location 'localhost:7007/api/orchestrator/v2/workflows/instances/<Your Workflow Instance Id>/result' \
--data ''

**Result - Assessment Workflow**
image

**Result - Non-assessment(e.g. Infrastructure) Workflow**
image

**Local Build - Feb 7th 10:15am**
image

**Local Test - Feb 7th 10:15am**
image

**Local TSC - Feb 7th 10:15am**
image

**Local Prettier - Feb 7th 10:15am**
image

### CODE REVIEW

File: plugins/orchestrator-backend/src/service/handlers.ts [hunk 1]
```diff
@@ -84,6 +85,36 @@ export async function executeWorkflowByIdV1(
...
+  }
+  if (!dataIndexService) {
+    throw new Error(
+      `No data index service provided for executing workflow with id ${instanceId}`,
+    );
+  }
+
+  const instanceResult = await getInstancesByIdV1(dataIndexService, instanceId);
+
+  if (instanceResult?.category !== WorkflowCategory.ASSESSMENT) {
```
Comment (gciavarrini) [inline plugins/orchestrator-backend/src/service/handlers.ts:103]: why this?
If `category = Infrastructure` the results are available. This should work for both infrastructure and assessment type. no?

Comment (rhkp) [reply plugins/orchestrator-backend/src/service/handlers.ts:103]: Okay we can remove that check completely.

Comment (rhkp) [reply plugins/orchestrator-backend/src/service/handlers.ts:103]: Done.

**Comment by gciavarrini** [comment]
Cherry-picked here https://github.com/janus-idp/backstage-plugins/pull/1182/commits/566b4b0d2ce26663be051443f31720ffa93bef32 

Thanks @rhkp :)

**Pull request closed without merging**